### ISSUE TITLE
Mod interaction with Ars Nouveau

### ISSUE THREAD

**Issue by iilonmasc** [wendall911/TinkerSurvival#79]
Hi there,
I am using tinkers survival with ars nouveau in a modpack and I am encountering a strange issue:
I first was not able to do any damage with Harm spells, so I added `are_nouveau` to the whitelisted mods, which worked for damaging spells.
However, using the Break spell (breaks blocks) I either get the childs laughter (spell book in offhand) or a message "A <tool> is required" (spell book in main hand) depending on the block I break (axe - planks, pickaxe - stone, sharp - wool)

I contacted the developer from Ars Nouveau and they told me the Break spell basically uses a diamond pickaxe for breaking stuff. So I added `building-minecraft:diamond_pickaxe` to the allowed items whitelist, but to no avail. I am still unable to break blocks with the break spell.

### Steps to reproduce:

Minecraft Version: 1.18.2
Forge: 40.1.73
Tinkers Survival: 2.3.9.1
Ars Nouveau: 2.7.4

1. Login to a world (with cheats and creative enabled)
2. Place a block of sandstone, wool, stone or planks
3. Get a creative spellbook `/give playername ars_nouveau:creative_spell_book`
4. Put the book in your hotbar, select it and press `C` to open the book
5. Create a `Projectile > Break` spell and use it (right click with book in main- or off-hand) 
6. -> Child laughter
7. Add `ars_nouveau` to whitelisted mods in tinkers survival config
8. Try the spell again (in main hand)
9. -> "A <tool> is required" message appears.
10. Try adding `building-minecraft:diamond_pickaxe` to allowed items (according to ars nouveau dev, the break spell uses a diamond pickaxe for breaking) and cast the spell again
11. -> Same message as before

### What should happen instead?

At least after adding either the mod, or `building-minecraft:diamond_pickaxe` the break spell should work.

**Comment by iilonmasc** [comment]
I just noticed something strange: I tried adding `building-ars_nouveau:novice_spell_book` to the whitelisted items, just in case. And apparently `itemidValidator` (https://github.com/wendall911/TinkerSurvival/blob/1.18.2/src/main/java/tinkersurvival/config/ConfigHandler.java#L123) does not allow underscore symbols in it's parsing regex for mod ids. The `modidValidator` does. I don't know how to test this or if this would change anything at all, I just noticed this weird behavior while testing this issue more.

**Comment by wendall911** [comment]
Thanks for the report. I'll figure out what the issue is. It is a little tricky on attacks not directly associated with the attacker. I'll likely have a fix out for this in the next day or so.

**Comment by wendall911** [comment]
Resolved in 2.3.9.2. https://www.curseforge.com/minecraft/mc-mods/tinkersurvival/files/3947176

I had to add a new type of tool for the spell book. Just need to add ars_nouveau to the whitelist and everything works as expected. Again, thanks for reporting!

Another side note, I have an integration mod that adds integration for Ars Nouveau here: https://www.curseforge.com/minecraft/mc-mods/tcintegrations
